PySDM/storages/common/backend.py

```diff
@@ -0,0 +1,273 @@
+"""
+The backend base classes
+
+The backend base classes are used to define the interface of the backends.
+The backends are responsible for the actual implementation of the storage operations.
+"""
+from typing import Protocol, Union
+
+import numpy as np
+
+from PySDM.storages.common.storage import Index, PairIndicator, Storage
+from PySDM.storages.thrust_rtc.conf import trtc
+
+DataType = Union[np.ndarray, trtc.DVVector]
+
+
+class Backend(Protocol):
+    """
+    The backend base class.
+
+    The backend base class is used to define the interface of the backends.
+    """
+
+    ...
```

Comment:
It seems to me that there is no point in having a `Protocol` class here. The `Backend` class doesn't possess any attributes or methods, so any class will be recognized as `Backend` by static type checkers.